firmware/baseband/proc_btlerx.cpp

```diff
@@ -26,245 +27,476 @@
+//     payload_type_5->WinOffset = ( (payload_type_5->WinOffset << 8) | payload_byte[20] );
+
+//     //Interval = reorder_bytes_str( payload_bytes((2*6+2*6+2*4+2*3+2*1+2*2+1):(2*6+2*6+2*4+2*3+2*1+2*2+2*2)) );
+//     payload_type_5->Interval = ( payload_byte[23] );
+//     payload_type_5->Interval = ( (payload_type_5->Interval << 8) | payload_byte[22] );
+
+//     //Latency = reorder_bytes_str( payload_bytes((2*6+2*6+2*4+2*3+2*1+2*2+2*2+1):(2*6+2*6+2*4+2*3+2*1+2*2+2*2+2*2)) );
+//     payload_type_5->Latency = ( payload_byte[25] );
+//     payload_type_5->Latency = ( (payload_type_5->Latency << 8) | payload_byte[24] );
+
+//     //Timeout = reorder_bytes_str( payload_bytes((2*6+2*6+2*4+2*3+2*1+2*2+2*2+2*2+1):(2*6+2*6+2*4+2*3+2*1+2*2+2*2+2*2+2*2)) );
+//     payload_type_5->Timeout = ( payload_byte[27] );
+//     payload_type_5->Timeout = ( (payload_type_5->Timeout << 8) | payload_byte[26] );
+
+//     //ChM = reorder_bytes_str( payload_bytes((2*6+2*6+2*4+2*3+2*1+2*2+2*2+2*2+2*2+1):(2*6+2*6+2*4+2*3+2*1+2*2+2*2+2*2+2*2+2*5)) );
+//     payload_type_5->ChM[0] = payload_byte[32];
+//     payload_type_5->ChM[1] = payload_byte[31];
+//     payload_type_5->ChM[2] = payload_byte[30];
+//     payload_type_5->ChM[3] = payload_byte[29];
+//     payload_type_5->ChM[4] = payload_byte[28];
+
+//     //tmp_bits = payload_bits((end-7) : end);
+//     //Hop = num2str( bi2de(tmp_bits(1:5), 'right-msb') );
+//     //SCA = num2str( bi2de(tmp_bits(6:end), 'right-msb') );
+//     payload_type_5->Hop = (payload_byte[33]&0x1F);
+//     payload_type_5->SCA = ((payload_byte[33]>>5)&0x07);
+//    }
+//     else
+//     {
+//    //TODO: Handle Unknown PDU.
+//      payload_type_R = (ADV_PDU_PAYLOAD_TYPE_R *)adv_pdu_payload;
+//      memcpy(payload_type_R->payload_byte, payload_byte, num_payload_byte);
+//     return(-1);
+//    }
+
+    return 0;
+}
+
 void BTLERxProcessor::execute(const buffer_c8_t& buffer) {
     if (!configured) return;
 
-    // FM demodulation
+    // Pulled this implementation from channel_stats_collector.c to time slice a specific packet's dB.
+    uint32_t max_squared = 0;
 
-    /*const auto decim_0_out = decim_0.execute(buffer, dst_buffer);
-        const auto channel = decim_1.execute(decim_0_out, dst_buffer);
+    void* src_p = buffer.p;
 
-        feed_channel_stats(channel);
+    while (src_p < &buffer.p[buffer.count]) {
+        const uint32_t sample = *__SIMD32(src_p)++;
+        const uint32_t mag_sq = __SMUAD(sample, sample);
+        if (mag_sq > max_squared) {
+            max_squared = mag_sq;
+        }
+    }
 
-        auto audio_oversampled = demod.execute(channel, work_audio_buffer);*/
+    const float max_squared_f = max_squared;
+    const int32_t max_dB = mag2_to_dbv_norm(max_squared_f * (1.0f / (32768.0f * 32768.0f)));
 
-    const auto decim_0_out = decim_0.execute(buffer, dst_buffer);
-    feed_channel_stats(decim_0_out);
+    decim_0.execute(buffer, dst_buffer);
+    feed_channel_stats(dst_buffer);
```

Comment:
I only see decim_0 used here, but you configured decim_1 and the demodulator.
Did you want to execute decim_1 (and process audio) too?

Reply:
Nope I forgot to remove that lol.